Toolchain/BuildGDB.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+set -e
+
+GDB_VERSION="13.1"
+GDB_MD5SUM="4aaad768ff2585464173c091947287ec"
+GDB_NAME="gdb-$GDB_VERSION"
+GDB_PKG="${GDB_NAME}.tar.xz"
+GDB_BASE_URL="https://ftp.gnu.org/gnu/gdb"
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+ARCH=${1:-"x86_64"}
+TARGET="$ARCH-pc-serenity"
+PREFIX="$DIR/Local/$ARCH-gdb"
+
+echo "Building GDB $GDB_VERSION for $TARGET"
+
+MD5SUM="md5sum"
+NPROC="nproc"
+
+SYSTEM_NAME="$(uname -s)"
+
+if [ "$SYSTEM_NAME" = "OpenBSD" ]; then
+    MD5SUM="md5 -q"
+    NPROC="sysctl -n hw.ncpuonline"
+    export CC=egcc
+    export CXX=eg++
+    export with_gmp=/usr/local
+    export LDFLAGS=-Wl,-z,notext
+elif [ "$SYSTEM_NAME" = "FreeBSD" ]; then
+    MD5SUM="md5 -q"
+    NPROC="sysctl -n hw.ncpu"
+    export with_gmp=/usr/local
+    export with_mpfr=/usr/local
+elif [ "$SYSTEM_NAME" = "Darwin" ]; then
```

Comment:
We should really move all this junk to a helper script file at some point. we've got it in at least 4 scripts in this directory.